pkg/gojail/param.go

```diff
@@ -0,0 +1,160 @@
+package gojail
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"reflect"
+	"strings"
+	"syscall"
+	"unsafe"
+)
+
+//JailParseParametersToIovec parses a map[string]interface{} parameter set to []syscall.Iovec
+//for use in Jail syscalls requiring []syscall.Iovec
+//Byte slices & pointers are considered in/out variables and will be filled with JailGet.
+//for setting handing use strings or ints instead.
+//gojail uses errmsg, and will error out if it's passed as a key
+//No validation is done w.r.t. the type required by the jail parameter
+func JailParseParametersToIovec(parameters map[string]interface{}) ([]syscall.Iovec, error) {
+	iovecs := make([]syscall.Iovec, 0)
+	for key, value := range parameters {
+		if key == "errmsg" {
+			return nil, errors.New("Usage of errmsg is reserved by gojail")
+		}
+		parIovec, err := paramToIOVec(key, value)
+		if err != nil {
+			return nil, err
+		}
+		//we allow vnet=false to return nil, nil continue in that case
+		if parIovec == nil {
+			continue
+		}
+		iovecs = append(iovecs, parIovec...)
+	}
+	return iovecs, nil
+}
+
+func splitParamInPrefixAndParam(p string) (prefix string, param string) {
+	s := strings.Split(p, ".")
+	param = s[len(s)-1]
+	s = s[:len(s)-1]
+	prefix = strings.Join(s, ".") + "."
+	return prefix, param
+}
+
+func paramToIOVec(key string, value interface{}) ([]syscall.Iovec, error) {
```

Comment:
Similar to my comment about `map[string]interface{}` above, my preference would be to avoid reflection-based logic for constructing the `syscall.Iovec` structs.  It is probably a bit more tedious, but I'd prefer to see structs like `JailCreateParams` that are able to serialize themselves into the appropriate `[]syscall.Iovec`.

For example:

```go
type JailCreateParams struct {
	Name    string
	Path    string
	Persist bool
}
```

This struct could implement an interface like this:

```go
type JailIovecSerializer interface {
	SerializeIovec() ([]syscall.Iovec, error)
}
```

```go
func (j *JailCreateParams) SerializeIovec() ([]syscall.Iovec, error) {
	iovec := make([]syscall.Iovec, 0)
	
	nameKey, err := syscall.ByteSliceFromString("name")
	if err != nil {
		return nil, err
	}
	nameVal, err := syscall.BytePtrFromString(j.Name)
	if err != nil {
		return nil, err
	}
	iovec = append(iovec, makeJailIovec(nameKey, nameVal, len(j.Name)+1)...)

	// repeat for other fields

	return iovec, nil
}
```